app/components/ChartSettings/modules/Metadata.js

```diff
@@ -39,10 +51,13 @@ export default class Metadata extends Component {
   };
 
   render() {
+    if (!this.shouldShowMetadata.subtitle) {
+      delete this.state.subtitle;
```

Comment:
You should always use `setState()` instead of directly modifying `this.state`; and never mess with the state inside the `render()` method. Instead, it looks like you could check like this below when you loop through the keys before rendering the `<DispatchField />`
```
Object.keys(this.state).map((key) => {
  return this.shouldShowMetadata[key] ? (
    // render JSX here
  ) : '';
})